src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -45,13 +45,10 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
 
             CSharpCompilation compilation = this.DeclaringCompilation;
 
-            // Assume that someone checked earlier that the attribute ctor is available and has no use-site errors.
             AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
 
-            // TODO (tomat): do we need to emit dynamic attribute on any synthesized field?
-            if (this.Type.ContainsDynamic())
+            if (this.Type.ContainsDynamic() && compilation.HasDynamicEmitAttributes())
             {
-                // Assume that someone checked earlier that the attribute ctor is available and has no use-site errors.
                 AddSynthesizedAttribute(ref attributes, compilation.SynthesizeDynamicAttribute(this.Type, customModifiersCount: 0));
```

Comment:
I think we should also fix the customModifiersCount on this call. We shouldn't assume it is always 0, we should get the actual value from the symbol..